engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/IsValueOfTypeNode.java

```diff
@@ -66,6 +67,12 @@ boolean doBigIntegerCheck(Type expectedType, EnsoBigInteger value) {
     return checkParentTypes(numbers.getBigInteger(), expectedType);
   }
 
+  @Specialization
+  boolean doUresolvedSymbol(Type expectedType, UnresolvedSymbol value) {
+    var funTpe = EnsoContext.get(this).getBuiltins().function();
+    return isSameObject.execute(expectedType, funTpe);
```

Comment:
`expectedType == funTpe` would be functionally equivalent and save us another `@Specialization` inside of the `isSameObject` node. When we know that both objects are `Type` we don't have to _"forget that"_ and ask `IsSameObjectNode` again...